Userland/Applications/PixelPaint/Tools/TextTool.cpp

```diff
@@ -0,0 +1,295 @@
+                // We are highlighting the final line of the selection.
+                // Start from first char and continue to selection end.
+                start_col = 0;
+                end_col = m_text_editor->selection().normalized().end().column();
+            } else {
+                // We are between the start and end lines, highlight the whole thing.
+                start_col = 0;
+                end_col = m_text_editor->line(i).length();
+            }
+            auto line_selection_length = end_col - start_col;
+            auto selected_string = m_text_editor->line(i).view().substring_view(start_col, line_selection_length);
+            auto text_before_selection = m_text_editor->line(i).view().substring_view(0, start_col);
+            auto selected_width = m_selected_font->width(selected_string);
+            auto selection_x_offset = m_selected_font->width(text_before_selection);
+
+            // the + 4 here is because that's how Painter::do_draw_text calculates line height, instead of asking
+            // the font it's preferred line height. If we don't replicate that here, the letters jump around when they
+            // get selected.
+            auto selection_y_offset = static_cast<int>((m_selected_font->pixel_size() + 4) * i);
+
+            auto selection_rect = Gfx::IntRect(selection_x_offset, selection_y_offset, selected_width, m_selected_font->preferred_line_height());
+            text_painter.fill_rect(selection_rect, m_text_editor->palette().selection());
+            text_painter.draw_text(selection_rect, selected_string, Gfx::TextAlignment::TopLeft, m_text_editor->palette().selection_text());
+        }
+    }
+
+    auto scaled_width = static_cast<int>(ceilf(m_editor->scale() * static_cast<float>(text_bitmap->width())));
+    auto scaled_height = static_cast<int>(ceilf(m_editor->scale() * static_cast<float>(text_bitmap->height())));
+    auto scaled_rect = Gfx::IntRect(text_location.x(), text_location.y(), scaled_width, scaled_height);
+    scaled_rect.set_location({ text_location.x(), text_location.y() });
+    painter.draw_scaled_bitmap(scaled_rect, text_bitmap, text_bitmap->rect(), 1.0);
+
+    // marching ants box
+    auto right_padding = m_selected_font->width("  "sv);
+    m_ants_rect = Gfx::IntRect(text_location.translated(-4, -2), { scaled_rect.width() + 4 + right_padding, scaled_rect.height() + 4 });
+    m_editor->draw_marching_ants(painter, m_ants_rect);
+
+    // Draw the blinking cursor.
+    if (m_cursor_blink_state) {
+        auto editor_cursor_rect = m_text_editor->cursor_content_rect();
+
+        /* TextEditor starts left most at 3, for TextTool this ends up putting the cursor in the middle of the letter.
+         * Looks better if we treat 0 as left most here, so we just translate it to the left. */
+        editor_cursor_rect.translate_by(-3, 0);
+
+        /* ImageEditor scale is a float, but we are working with int and IntRects. */
```

Comment:
Nit-pick: Use `// ...` for comments, instead of `/* ... */`

Reply:
Should be fixed now with latest push.